### ISSUE TITLE
Newline in text array loses backslash

### ISSUE THREAD

**Issue by jehna** [brianc/node-postgres#1147]
### Issue:

When inserting/updating a column of type `text[]`, newlines get converted to a character `n`.
### How to reproduce:

Run the following code:

```
var pg = require('pg');
var client = new pg.Client();
client.connect(function (err) {
  client.query('CREATE TABLE test (array_test text[])', function(err, result) {
    client.query('INSERT INTO test (array_test) VALUES ($1)', [['hello\nworld']], function (err, result) {
      client.query('SELECT * FROM test', function (err, result) {
        console.log(result.rows[0]);
        client.end();
      })
    })
  })
});
```
#### Behaviour:

The console.log is missing a backslash:

```
{ array_test: [ 'hellonworld' ] }
```
#### Expected behaviour

The console.log should output a string including the backslash:

```
{ array_test: [ 'hello\nworld' ] }
```
### Quick fix

Quick fix for this issue is to change `/lib/utils.js:28` to:

```
result = result + JSON.stringify(prepareValue(val[i])).replace('\\n', '\n');
```
### Other notes:

Not related to #895 (which seems to be fixed atm)

**Comment by vitaly-t** [comment]
No bug there.

See http://stackoverflow.com/questions/935/string-literals-and-escape-characters-in-postgresql

**Comment by jehna** [comment]
@vitaly-t just to clarify: I'm not trying to insert escaped newlines (`\n`), but real newlines in JavaScript.

So changing `[['hello\nworld']]` to:

```
[[`hello
world`]]
```

results in the same bug:

```
{ array_test: [ 'hellonworld' ] }
```

**Comment by jehna** [comment]
Code for the clarification:

```
var pg = require('pg');
var client = new pg.Client();
var insertMe = `hello
world`;
client.connect(function (err) {
  client.query('CREATE TABLE test (array_test text[])', function(err, result) {
    client.query('INSERT INTO test (array_test) VALUES ($1)', [[insertMe]], function (err, result) {
      client.query('SELECT * FROM test', function (err, result) {
        console.log(result.rows[0]);
        client.end();
      })
    })
  })
});
```

**Comment by vitaly-t** [comment]
The first linked answered your question, it works that way, there's nothing else there.

**Comment by joskuijpers** [comment]
Does this problem only occur with text[], or also with text? I never had problems with the text type but haven't used text[]. Could you test that?

**Comment by jehna** [comment]
@joskuijpers Tested with the following code, the `text` seems to work just fine:

```
var pg = require('pg');
var client = new pg.Client();
var insertMe = `hello
world`;
client.connect(function (err) {
  client.query('CREATE TABLE test (array_test text)', function(err, result) {
    client.query('INSERT INTO test (array_test) VALUES ($1)', [insertMe], function (err, result) {
      client.query('SELECT * FROM test', function (err, result) {
        console.log(result.rows[0]);
        client.end();
      })
    })
  })
});
```

**Comment by davidskuza** [comment]
![image](https://cloud.githubusercontent.com/assets/10241901/19873231/28e3dc76-9fbe-11e6-8486-42efedf5a4b1.png)

**Comment by jehna** [comment]
@davidskuza You'll have to escape the string so it'd work:

```
postgres=# CREATE TABLE test (array_test text[]);
CREATE TABLE
postgres=# INSERT INTO test (array_test) values (e'{"hello\nworld"}');
INSERT 0 1
postgres=# SELECT * FROM test;
 array_test 
------------
 {"hello   +
 world"}
(1 row)
```

**Comment by vitaly-t** [comment]
My questions is - why Prepared Statements are affected by this also?

**Comment by charmander** [comment]
@vitaly-t, it’s a bug in the serializer; why wouldn’t it affect prepared statements? Also, this was fixed in 6.1.1, note.

**Comment by jehna** [comment]
Closing this issue from `node-postgres`, since the version `6.1.1` seems to work fine now.

Thank you @charmander
